pkgs/development/compilers/koka/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, cmake, mkDerivation, fetchFromGitHub, alex, array, base, bytestring, cond
+, containers, directory, extra, filepath, haskeline, hpack, hspec
+, hspec-core, json, lib, mtl, parsec, process, regex-compat, text, time }:
+
+let
+  version = "2.1.1";
+  src = fetchFromGitHub {
+    owner = "koka-lang";
+    repo = "koka";
+    rev = "v${version}";
+    sha256 = "sha256-cq+dljfTKJh5NgwQfxQQP9jRcg2PQxxBVEgQ59ll36o=";
+    fetchSubmodules = true;
+  };
+  kklib = stdenv.mkDerivation {
+    pname = "kklib";
+    inherit version;
+    src = "${src}/kklib";
+    nativeBuildInputs = [ cmake ];
+  };
+in
+mkDerivation rec {
+  pname = "koka";
+  inherit version src;
+  isLibrary = false;
+  isExecutable = true;
+  libraryToolDepends = [ hpack ];
+  executableHaskellDepends = [
+    array base bytestring cond containers directory haskeline mtl
+    parsec process text time cmake kklib
+  ];
+  executableToolDepends = [ alex ];
+  testHaskellDepends = [
```

Comment:
You can remove all these and the inputs as well as I don't think we will be able to run tests anytime soon as they need stack.